packages/editor/src/components/rich-text/index.native.js

```diff
@@ -96,11 +93,12 @@ export class RichText extends Component {
 
 	shouldComponentUpdate( nextProps ) {
 		// The check below allows us to avoid updating the content right after an `onChange` call
+		if ( nextProps.content.forceUpdate ) {
```

Comment:
Once you start passing down `attributes.content` as prop directly you can make it immutable and always create a new reference whenever it changes to simplify all those checks.